test/functional/wallet_migration.py

```diff
@@ -1012,6 +1014,140 @@ def check_comments():
 
         wallet.unloadwallet()
 
+    def test_p2wsh(self):
+        self.log.info("Test that P2WSH output scripts are migrated")
+        def_wallet = self.nodes[0].get_wallet_rpc(self.default_wallet_name)
+
+        wallet = self.create_legacy_wallet("p2wsh")
+
+        # Craft wsh(pkh(key))
+        pubkey = wallet.getaddressinfo(wallet.getnewaddress())["pubkey"]
+        pkh_script = key_to_p2pkh_script(pubkey).hex()
+        wsh_pkh_script = script_to_p2wsh_script(pkh_script).hex()
+        wsh_pkh_addr = script_to_p2wsh(pkh_script)
+
+        wallet.importaddress(address=pkh_script, p2sh=False)
+        wallet.importaddress(address=wsh_pkh_script, p2sh=False)
+
+        def_wallet.sendtoaddress(wsh_pkh_addr, 5)
+        self.generate(self.nodes[0], 6)
+        assert_equal(wallet.getbalances()['mine']['trusted'], 5)
+
+        wallet.migratewallet()
```

Comment:
We generally assume that test readers have a basic understanding of how RPCs work, such as that unless stated otherwise in their help docs, all RPCs throw when they fail.

Reply:
Since this PR needed to be rebased for #31248 which added a `migrate_and_get_rpc` helper function that replaced all `migratewallet` calls, I've added a check to that function to verify that the wallet is legacy before migration, and descriptors after.